delta/plugins/composite-views/src/main/scala/ch/epfl/bluebrain/nexus/delta/plugins/compositeviews/model/CompositeViewRejection.scala

```diff
@@ -0,0 +1,113 @@
+import ch.epfl.bluebrain.nexus.delta.sdk.model.projects.ProjectRef
+import ch.epfl.bluebrain.nexus.delta.sourcing.processor.AggregateResponse.EvaluationError
+import io.circe.Json
+
+/**
+  * Enumeration of composite view rejection types.
+  *
+  * @param reason a descriptive message as to why the rejection occurred
+  */
+sealed abstract class CompositeViewRejection(val reason: String) extends Product with Serializable
+
+object CompositeViewRejection {
+
+  /**
+    * Signals a rejection caused by an attempt to create or update an composite view with a permission that is not
+    * defined in the permission set singleton.
+    *
+    * @param permission the provided permission
+    */
+  final case class PermissionIsNotDefined(permission: Permission)
+      extends CompositeViewRejection(
+        s"The provided permission '${permission.value}' is not defined in the collection of allowed permissions."
+      )
+
+  /**
+    * Rejection returned when attempting to create a view with an id that already exists.
+    *
+    * @param id the view id
+    */
+  final case class ViewAlreadyExists(id: Iri, project: ProjectRef)
+      extends CompositeViewRejection(s"Composite view '$id' already exists in project '$project'.")
+
+  /**
+    * Rejection returned when a view that doesn't exist.
+    *
+    * @param id the view id
+    */
+  final case class ViewNotFound(id: Iri, project: ProjectRef)
+      extends CompositeViewRejection(s"Composite view '$id' not found in project '$project'.")
+
+  /**
+    * Rejection returned when attempting to update/deprecate a view that is already deprecated.
+    *
+    * @param id the view id
+    */
+  final case class ViewIsDeprecated(id: Iri) extends CompositeViewRejection(s"Composite view '$id' is deprecated.")
+
+  /**
+    * Rejection returned when a subject intends to perform an operation on the current view, but either provided an
+    * incorrect revision or a concurrent update won over this attempt.
+    *
+    * @param provided the provided revision
+    * @param expected the expected revision
+    */
+  final case class IncorrectRev(provided: Long, expected: Long)
+      extends CompositeViewRejection(
+        s"Incorrect revision '$provided' provided, expected '$expected', the view may have been updated since last seen."
+      )
+
+  /**
+    * Rejection returned when a subject intends to retrieve a view at a specific revision, but the provided revision
+    * does not exist.
+    *
+    * @param provided the provided revision
+    * @param current  the last known revision
+    */
+  final case class RevisionNotFound(provided: Long, current: Long)
+      extends CompositeViewRejection(
+        s"Revision requested '$provided' not found, last known revision is '$current'."
+      )
+
+  /**
+    * Rejection returned when the project for a [[CrossProjectSource]] does not exist.
+    */
+  final case class CrossProjectSourceProjectNotFound(crossProjectSource: CrossProjectSource)
```

Comment:
It would be useful to have something like:

```scala
sealed trait CompositeViewSourceRejection extends CompositeViewRejection

final case class CrossProjectSourceProjectNotFound(...) extends CompositeViewRejection
final case class CrossProjectSourceForbidden(...) extends CompositeViewRejection
final case class InvalidRemoteProjectSource(...) extends CompositeViewRejection
```

Same goes for projections